packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -241,6 +246,101 @@ abstract class Action<T extends Intent> with Diagnosticable {
 /// If you listen to an [Action] widget in a widget hierarchy, you should use
 /// this widget. If you are using an [Action] outside of a widget context, then
 /// you must call removeListener yourself.
+///
+/// {@tool dartpad --template=stateful_widget_scaffold_center}
+/// This example shows how ActionListener handles adding and removing of
+/// the [listener] in the widget lifecycle.
+///
+/// ```dart preamble
+/// class ExampleActionListener extends StatefulWidget {
```

Comment:
Nit: What do you think about changing this name to `ActionListenerExample`?  I think it's more clear that it is an example about ActionListener, and not an ActionListener itself.  Not a big deal, though.